Sources/KlaviyoCore/Networking/NetworkSession.swift

```diff
@@ -35,10 +35,39 @@ public struct NetworkSession {
 
     public static let networkTimeout: UInt64 = 10_000_000_000 // in nanoseconds (10 seconds)
 
+    static func getPluginConfiguration(bundle: Bundle = .main) -> (name: String, version: String)? {
+        guard let plistURL = bundle.url(forResource: "klaviyo-plugin-configuration", withExtension: "plist") else {
+            return nil
+        }
+
+        guard let plistData = try? Data(contentsOf: plistURL) else {
+            return nil
+        }
+
+        guard let plist = try? PropertyListSerialization.propertyList(from: plistData, options: [], format: nil) as? [String: Any] else {
+            return nil
+        }
+
+        guard let pluginName = plist["klaviyo_sdk_plugin_name_override"] as? String else {
+            return nil
+        }
+        guard let pluginVersion = plist["klaviyo_sdk_plugin_version_override"] as? String else {
+            return nil
+        }
+
+        return (name: pluginName, version: pluginVersion)
+    }
+
     public static let defaultUserAgent = { () -> String in
         let appContext = environment.appContextInfo()
         let klaivyoSDKVersion = "klaviyo-\(environment.sdkName())/\(environment.sdkVersion())"
-        return "\(appContext.executable)/\(appContext.appVersion) (\(appContext.bundleId); build:\(appContext.appBuild); \(appContext.osVersionName)) \(klaivyoSDKVersion)"
+        var userAgent = "\(appContext.executable)/\(appContext.appVersion) (\(appContext.bundleId); build:\(appContext.appBuild); \(appContext.osVersionName)) \(klaivyoSDKVersion)"
+        
+        if let pluginConfig = getPluginConfiguration() {
```

Comment:
```suggestion
        // if the SDK is running through an Expo plugin, we want to append the plugin info
        if let pluginConfig = getPluginConfiguration() {
```
not sure if that's exactly the correct way to phrase what's going on but it would help to have a comment explaining why we're adding the plugin config